File: api/src/endpoint/repository/get_repository_preview.rs
use serde::{Deserialize, Serialize};

use super::default_ref;
use crate::{endpoint::Endpoint, resource::repository::RepositoryPreviewResource};

pub struct GetRepositoryPreview;

impl Endpoint for GetRepositoryPreview {
    const PATH: &'static str = "/repository/{owner}/{repo}/preview";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetRepositoryPreviewRequest;
    type Response = GetRepositoryPreviewResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetRepositoryPreviewRequest {
    #[serde(default = "default_ref")]
    pub ref_name: String,
    pub preview_lines: Option<u32>,
}

pub type GetRepositoryPreviewResponse = RepositoryPreviewResource;

File: gitdot-api-ts/src/endpoint/repository/get-repository-preview.ts
import { z } from "zod";

import { RepositoryPreviewResource } from "../../resource";

export const GetRepositoryPreviewRequest = z.object({
  ref_name: z.string(),
  preview_lines: z.number().int().optional(),
});
export type GetRepositoryPreviewRequest = z.infer<
  typeof GetRepositoryPreviewRequest
>;

export const GetRepositoryPreview = {
  path: "/repository/{owner}/{repo}/preview",
  method: "GET",
  request: GetRepositoryPreviewRequest,
  response: RepositoryPreviewResource,
} as const;
export type GetRepositoryPreview = typeof GetRepositoryPreview;
